src/libbson/tests/test-mcd-atomic.c

```diff
@@ -0,0 +1,128 @@
+      /* Compare-exchange fail: */                                             \
+      got = ATOMIC (Kind, compare_exchange_weak) (&value, 8, 12, MemOrder);    \
+      Assert (got, ==, 9);                                                     \
+      Assert (value, ==, 9);                                                   \
+      /* Compare-exchange-weak succeed: */                                     \
+      /* 'weak' may fail spuriously, so it must *eventually* succeed */        \
+      for (i = 0; i < 10000 && value != 53; ++i) {                             \
+         got = ATOMIC (Kind, compare_exchange_weak) (&value, 9, 53, MemOrder); \
+         Assert (got, ==, 9);                                                  \
+      }                                                                        \
+      /* Check that it evenutally succeeded */                                 \
+      Assert (value, ==, 53);                                                  \
+   } while (0)
+
+#define TEST_INTEGER_KIND(Kind, TypeName, Assert)                                 \
+   do {                                                                           \
+      TEST_KIND_WITH_MEMORDER (Kind, TypeName, mcd_memory_order_relaxed, Assert); \
+      TEST_KIND_WITH_MEMORDER (Kind, TypeName, mcd_memory_order_acq_rel, Assert); \
+      TEST_KIND_WITH_MEMORDER (Kind, TypeName, mcd_memory_order_acquire, Assert); \
+      TEST_KIND_WITH_MEMORDER (Kind, TypeName, mcd_memory_order_release, Assert); \
+      TEST_KIND_WITH_MEMORDER (Kind, TypeName, mcd_memory_order_consume, Assert); \
+      TEST_KIND_WITH_MEMORDER (Kind, TypeName, mcd_memory_order_seq_cst, Assert); \
+   } while (0)
+
+
+static void
+test_integers (void)
+{
+   TEST_INTEGER_KIND (int64, int64_t, ASSERT_CMPINT64);
+   TEST_INTEGER_KIND (int32, int32_t, ASSERT_CMPINT32);
+   TEST_INTEGER_KIND (int16, int16_t, ASSERT_CMPINT);
+   TEST_INTEGER_KIND (int8, int8_t, ASSERT_CMPINT);
+   TEST_INTEGER_KIND (int, int, ASSERT_CMPINT);
+}
+
+
+static void
+test_pointers (void)
+{
+   int u = 12;
+   int v = 9;
+   int w = 91;
+   int *ptr = &v;
+   int *other;
+   int *prev;
+   other = mcd_atomic_ptr_fetch ((void *) &ptr, mcd_memory_order_relaxed);
+   ASSERT_CMPVOID (other, ==, ptr);
+   prev = mcd_atomic_ptr_exchange ((void *) &other, &u, mcd_memory_order_relaxed);
+   ASSERT_CMPVOID (prev, ==, &v);
+   ASSERT_CMPVOID (other, ==, &u);
+   prev = mcd_atomic_ptr_compare_exchange_strong ((void *) &other, &v, &w, mcd_memory_order_relaxed);
+   ASSERT_CMPVOID (prev, ==, &u);
+   ASSERT_CMPVOID (other, ==, &u);
+   prev = mcd_atomic_ptr_compare_exchange_strong ((void *) &other, &u, &w, mcd_memory_order_relaxed);
+   ASSERT_CMPVOID (prev, ==, &u);
+   ASSERT_CMPVOID (other, ==, &w);
+}
+
+
+static void
+test_thread_fence (void)
+{
+   mcd_atomic_thread_fence ();
+}
+
+static void
+test_thrd_yield (void)
+{
+   mcd_thrd_yield ();
+}
+
+void
+test_atomic_install (TestSuite *suite)
```

Comment:
```suggestion
test_mcd_atomic_install (TestSuite *suite)
```
Avoid conflict with `test-atomic.c` + add to `test-libmongoc-main.c`.